docs/custom_node_development.md

```diff
@@ -0,0 +1,125 @@
+# Custom Node Development Guide
+
+## Overview
+
+Custom node in OpenVINO Model Server simplifies linking deep learning models into a complete pipelines even if the inputs and output
+of the sequential models do not fit. In many cases, output of one model can not be directly passed to another one.
+The data might need to be analysed, filtered or converted to different format. Those operations can not be easily implemented
+in AI frameworks or are simply not supported. Custom node addresses this challenge. They allows employing a dynamic library
+developed in C++ or C to perform arbitrary data transformations. 
+
+## Custom Node API
+
+The custom node library must implement the API interface defined in [custom_node_interface.h](../src/custom_node_interface.h).
+The interface is defined in `C` to simplify compatibility with various compilers. The library could use third party components
+linked statically or dynamically. OpenCV is a built in component in OVMS which could be used to perform manipulation on the image
```

Comment:
`linked statically or dynamically' this means that we have to test both cases

Reply:
we don't have tests for static linking but it should be doable if the namespace is not conflicting.